src/main/java/org/spongepowered/mod/mixin/core/event/player/MixinEventPlayerInteractBlock.java

```diff
@@ -79,4 +85,15 @@ public EntityInteractionType getInteractionType() {
         return Optional.fromNullable(new Cause(null, this.entityPlayer, null));
     }
 
+    @Override
+    public Optional<Vector3d> getClickedPosition() {
+        return Optional.of(VecHelper.toVector(this.pos).toDouble());
```

Comment:
Well it definitely worked when I tried it. `javax.vecmath.Vector3d` is the wrong class, use `com.flowpowered.math.vector.Vector3d`
But you are correct in using `@Redirect`, with redirect you can replace existing code which is what needs to be done.
When it comes to mixins sometimes it can be quite messy but there's almost always a way.